fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgresql/PostgreSqlResourceNoProcDAO.java

```diff
@@ -0,0 +1,541 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.postgresql;
+
+import static com.ibm.fhir.persistence.jdbc.JDBCConstants.UTC;
+
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.List;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.transaction.TransactionSynchronizationRegistry;
+
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceVersionIdMismatchException;
+import com.ibm.fhir.persistence.jdbc.FHIRPersistenceJDBCCache;
+import com.ibm.fhir.persistence.jdbc.connection.FHIRDbFlavor;
+import com.ibm.fhir.persistence.jdbc.dao.api.FhirRefSequenceDAO;
+import com.ibm.fhir.persistence.jdbc.dao.api.IResourceReferenceDAO;
+import com.ibm.fhir.persistence.jdbc.dao.api.JDBCIdentityCache;
+import com.ibm.fhir.persistence.jdbc.dao.api.ParameterDAO;
+import com.ibm.fhir.persistence.jdbc.dao.impl.JDBCIdentityCacheImpl;
+import com.ibm.fhir.persistence.jdbc.dao.impl.ParameterVisitorBatchDAO;
+import com.ibm.fhir.persistence.jdbc.dao.impl.ResourceDAOImpl;
+import com.ibm.fhir.persistence.jdbc.dto.ExtractedParameterValue;
+import com.ibm.fhir.persistence.jdbc.dto.Resource;
+import com.ibm.fhir.persistence.jdbc.exception.FHIRPersistenceDBConnectException;
+import com.ibm.fhir.persistence.jdbc.exception.FHIRPersistenceDataAccessException;
+import com.ibm.fhir.persistence.jdbc.exception.FHIRPersistenceFKVException;
+import com.ibm.fhir.persistence.jdbc.impl.ParameterTransactionDataImpl;
+import com.ibm.fhir.persistence.jdbc.util.ResourceTypesCache;
+
+/**
+ * Data access object for writing FHIR resources to a PostgreSQL database.
+ * Like the Derby implementation, this version uses individual SQL statements
+ * instead of a stored procedure and is useful for debugging performance and/or
+ * concurrency issues. It should not be used in production scenarios because
+ * the numerous app-database round-trips slow things down considerably.
+ * 
+ * @implNote This class follows the logic of the stored procedure, but does so
+ * using a series of individual JDBC statements.
+ */
+public class PostgreSqlResourceNoProcDAO extends ResourceDAOImpl {
```

Comment:
Inconsistent naming

Reply:
Why are we splitting the implementations? is one better than the other? 
Should we add experimental or test to the package?

Reply:
Just following previously established convention (Albert's initial implementation)